src/main/java/com/oath/halodb/HaloDBInternal.java

```diff
@@ -312,6 +312,14 @@ void setIOErrorFlag() throws IOException {
 
     void pauseCompaction() throws IOException {
         compactionManager.pauseCompactionThread();
```

Comment:
Got to check HaloDB after a long time and noticed this PR. 

`pauseCompactionThread` already waits for compaction thread to stop before returning. 
https://github.com/yahoo/HaloDB/blob/b81eeacc9dddd7ceb14eef7bbe726550d320dbc5/src/main/java/com/oath/halodb/CompactionManager.java#L66

Therefore is this change really required?

Reply:
Got point. Yes. I didn't see that . Will double confirm by checking production log  and close this PR

Reply:
@amannaly You are right. Thanks!
```
=2019-08-13 01:32:19,540  INFO [pool-5-thread-25] (com.oath.halodb.CompactionManager) - Pausing compaction thread ...
=2019-08-13 01:32:19,545  INFO [CompactionThread] (com.oath.halodb.CompactionManager) - Compaction thread stopped.
```